packages/testing/ts/types.ts

```diff
@@ -0,0 +1,136 @@
+import { Keypair, PrivKey } from "maci-domainobjs";
+import { genMaciStateFromContract } from "maci-sdk";
+
+import type { Signer } from "ethers";
+import type {
+  MACI,
+  Verifier,
+  VkRegistry,
+  FreeForAllGatekeeper,
+  ConstantInitialVoiceCreditProxy,
+  IIpfsMessage,
+} from "maci-sdk";
+
+/**
+ * A util interface that represents a vote object
+ */
+export interface IVote {
+  voteOptionIndex: bigint;
+  voteWeight: bigint;
+  nonce: bigint;
+}
+
+/**
+ * A util interface that represents a briber object
+ */
+export interface IBriber {
+  voteOptionIndices: bigint[];
+}
+
+/**
+ * A util interface that represents a change user keys object
+ */
+export interface IChangeUsersKeys {
+  voteOptionIndex: bigint;
+  voteWeight: bigint;
+}
+
+/**
+ * A util interface that represents a test suite
+ */
+export interface ITestSuite {
+  name: string;
+  description: string;
+  numVotesPerUser: number;
+  numUsers: number;
+  expectedTally: number[];
+  expectedSpentVoiceCredits: number[];
+  expectedTotalSpentVoiceCredits: number;
+  bribers?: IBriber[];
+  votes?: IVote[][];
+  changeUsersKeys?: IChangeUsersKeys[][];
+}
+
+/**
+ * A util interface that represents deployed test contracts
+ */
+export interface IDeployedTestContracts {
+  maci: MACI;
+  verifier: Verifier;
+  vkRegistry: VkRegistry;
+  gatekeeper: FreeForAllGatekeeper;
+  initialVoiceCreditProxy: ConstantInitialVoiceCreditProxy;
+}
+
+/**
+ * A util interface that represents testing data from the TestingClass object
+ */
+export interface IContractsData {
+  initialized: boolean;
+  users?: User[];
+  maciContractAddress?: string;
+  maciState?: Awaited<ReturnType<typeof genMaciStateFromContract>>;
+  polls?: string[];
+}
+
+export interface IRelayTestMessagesArgs {
+  messages: IIpfsMessage[];
+  maciAddress: string;
+  signer: Signer;
+  pollId: number;
+}
+
+/**
+ * A class representing a user and its votes
+ */
+export class User {
```

Comment:
Let's move this to a separate file

Reply:
if I move that to a separate file, then I will need to move out something else to another file to avoid an import cycle - either move IVote too or move IContractsData - what you think it's best?

Reply:
So to prevent dependency cycle, you can use `import type`. All the types can have cycles and in the result bundle (js) types are not available. So it's not a problem if you're using `import type` for `IVote` or `IContractsData`.

Reply:
oh nice thanks